### ISSUE TITLE
Guides of some sort

### ISSUE THREAD

**Issue by prokoudine** [friction2d/friction#250]
Not sure of this is on the table before 1.0 is out, but something I find rather lacking are guides or any other way to align things. This is frustrating to the point where I need to design the layout in Inkscape, with all the guides baked in as lines, export to PNG, then import it as a background bitmap.

**Comment by rodlie** [comment]
The plan was to set a feature freeze for 1.0 by the end of this week, so I doubt this will make it.

This also need some planning, what is the MVP for this feature?

**Comment by prokoudine** [comment]
If you go the route if implementing conventional guides rather than smart guides, I'd probably expect the MVP to do this:

1. View > Guides > Add new guide / Remove all guides
2. "Add new guides" dialog allows choosing orientation (vertical or horizontal) and setting the exact position in pixels
3. Guides can be dragged up/down or left/right
4. No snapping in the MVP, just place a guide where it should be and align objects manually looking at the bounding box or pivot point
5. "Remove all guides" just nukes all guides from the orbit without asking for confirmation, but it should be undoable.

But frankly, if you are planning to do a feature freeze so that you could open new development later on, I'd suggest skipping this and — once 1.0 is out — starting to work on something better.

**Comment by rodlie** [comment]
Sounds like a plan.

Milestone v1.1 seems like the best option. I'm not adding new major features to `main` anymore, need to fix/cleanup what we have.

**Comment by AlexKiryanov** [comment]
@rodlie , good day!
@prokoudine , good day! Glad to read you!
Support the idea, a very necessary tool.
Also as an option, if you point to the guide and click 2 times on it, a window with coordinates will appear so that you can change it if necessary.

**Comment by kaixoo** [comment]
Inkscape supports smart snapping (somewhat), I should look at how they do it.

**Comment by prokoudine** [comment]
> Inkscape supports smart snapping (somewhat)

Inkscape has a very decent implementation, very nearly on par with Figma for what I do.

**Comment by samytichadou** [comment]
Agreed for the inkscape implementation ! Guides and snapping are a must have feature for motion graphics, can't wait for v1.1 to have these :)

**Comment by rodlie** [comment]
There are no guarantee that this will happen in v1.1, as every feature request usually is just added to the next feature milestone.

But I will try to make it a priority. It probably needs several feature releases to be "complete", but an initial implementation should be doable for v1.1.
